clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -272,6 +272,7 @@ private boolean trySend(long now) {
 
     private void clientPoll(long timeout, long now) {
         client.poll(timeout, now);
+        now = time.milliseconds();
```

Comment:
Removed now from clearUnsentRequests (thank you @hachikuji). Fixed pollUnsentRequests by updating now inside that call (thank you @ewencp). 

Adding a new Now class makes sense but will require a large cleanup patch over the entire code base. I wonder if it is worth opening a new Jira for that and following up (I can do that), while going ahead with the immediate fixes.